### ISSUE TITLE
tests/e2e: `TestLibvirtCreatePeerPodAndCheckEnvVariableLogsWithImageOnly` test unstable

### ISSUE THREAD

**Issue by stevenhorsman** [confidential-containers/cloud-api-adaptor#1831]
We see occasional (anecdotally <20% of the time) failures on the libvirt nightly CI, which seems to always (so far) pass on re-run and now we've seen in on a PR test, so it's becoming more of an obstacle, so we should investigate it when we get the chance
```
=== RUN   TestLibvirtCreatePeerPodAndCheckEnvVariableLogsWithImageOnly
=== RUN   TestLibvirtCreatePeerPodAndCheckEnvVariableLogsWithImageOnly/EnvVariablePeerPodWithImageOnly_test
    assessment_runner.go:262: timed out waiting for the condition
--- FAIL: TestLibvirtCreatePeerPodAndCheckEnvVariableLogsWithImageOnly (600.10s)
    --- FAIL: TestLibvirtCreatePeerPodAndCheckEnvVariableLogsWithImageOnly/EnvVariablePeerPodWithImageOnly_test (600.10s)
```

**Comment by stevenhorsman** [comment]
This is getting worse and we are hitting it multiple times on each PR now. I've tried running this test locally and in about 8 re-runs it worked every time, so I'm not sure of the cause of the failure. In the short term I think we need to skip it in the CI to stop it blocking PRs.

**Comment by stevenhorsman** [comment]
It is possible that this is related to the image-pull changes as Chengyu is touch the config merge code in https://github.com/kata-containers/kata-containers/pull/9695, so after this, we should try re-testing this.

**Comment by stevenhorsman** [comment]
Hmm, this is suspicious, now the e2e tests related to env are skipped I've seen:
```
=== RUN   TestLibvirtCreatePeerPodAndCheckWorkDirLogs
=== RUN   TestLibvirtCreatePeerPodAndCheckWorkDirLogs/WorkDirPeerPod_test
    assessment_runner.go:262: timed out waiting for the condition
--- FAIL: TestLibvirtCreatePeerPodAndCheckWorkDirLogs (600.16s)
    --- FAIL: TestLibvirtCreatePeerPodAndCheckWorkDirLogs/WorkDirPeerPod_test (600.16s)
```
start failing, so maybe it's related to something before now being cleaned up, or the workdir has the same issue?

**Comment by stevenhorsman** [comment]
> start failing, so maybe it's related to something before now being cleaned up, or the workdir has the same issue?

This has failed the last three nightlies, so I will raise a PR to skip this for now

**Comment by wainersm** [comment]
@stevenhorsman yesterday I ran TestLibvirtCreatePeerPodAndCheckWorkDirLogs a couple of times locally with the hope of reproducing the error but it always passed!

Then I started working on a golang equivalent of `kubectl describe` so we could print more info on CI, but ran out of time...

**Comment by stevenhorsman** [comment]
> @stevenhorsman yesterday I ran TestLibvirtCreatePeerPodAndCheckWorkDirLogs a couple of times locally with the hope of reproducing the error but it always passed!

Yeah - I have this experience with the other tests too. My hope is that a new version of the kata-agent and image-rs might have addressed some of these, so I will re-test after they've been bumped

**Comment by stevenhorsman** [comment]
In https://github.com/confidential-containers/cloud-api-adaptor/pull/2183 I've tried re-enabling all the tests and it seems that only
- TestLibvirtCreatePeerPodAndCheckEnvVariableLogsWithImageAndDeployment
- TestLibvirtCreatePeerPodWithLargeImage
seem to still be failing, but we'll need to keep a close eye on the stability of the rest if/when it gets merged.

**Comment by stevenhorsman** [comment]
Tracking of unstable tests:
- 13/12/24 nightly: TestLibvirtCreatePeerPodAndCheckEnvVariableLogsWithImageOnly failed on one of the test runs
